### PR TITLE
chore: @trezor/transport typescript and lint refactor

### PR DESCRIPTION
**Pull request by mroz22**
This PR contains almost no runtime changes. It focuses solely on typescript and linting changes in @trezor/transport package.

### CODE REVIEW

File: packages/integration-tests/projects/transport/common.setup.js [hunk 1]
```diff
@@ -19,9 +19,9 @@ const wait = ms =>
 
 const setup = async controller => {
     await controller.connect();
+    await controller.send({ type: 'bridge-stop' });
```
Comment (Nodonisko) [inline packages/integration-tests/projects/transport/common.setup.js:22]: just curious, why these changes? why adding stop and removing start?

File: packages/transport/.eslintrc.js [hunk 2]
```diff
@@ -2,11 +2,10 @@ module.exports = {
...
         'no-underscore-dangle': 'off',
-        '@typescript-eslint/no-use-before-define': 'off',
-        '@typescript-eslint/ban-types': 'off',
-        'require-await': 'off',
         'no-restricted-syntax': 'off',
         'no-await-in-loop': 'off',
-        'func-names': 'off',
+
+        // todo: maybe consider this rule globally?
+        quotes: ['error', 'single', { avoidEscape: true }],
```
Comment (Nodonisko) [inline packages/transport/.eslintrc.js]: I believe this rule shouldn't be necessary because we have prettier and that should do this automatically. We rather should introduce some prettier check for CI for whole code base. prettier-eslint seems to miss some things anyway

File: packages/transport/src/fallback.ts [hunk 3]
```diff
@@ -1,34 +1,26 @@
...
-    transports: Array<Transport>;
-    configured: boolean;
-    version: string;
-    debug = false;
-
-    // note: activeTransport is actually "?Transport", but
-    // everywhere I am using it is in `async`, so error gets returned as Promise.reject
+    _availableTransports: Array<Transport> = [];
+    activeName = '';
+    // @ts-ignore
```
Comment (Nodonisko) [inline packages/transport/src/fallback.ts:6]: why is this needed?

File: packages/transport/tsconfig.lib.json [hunk 4]
```diff
@@ -10,14 +10,8 @@
...
-    "exclude": [
-        "node_modules",
-        "**/tests/**",
-        "**/fixtures/**",
-        "**/*.test.ts",
-        "**/e2e/**",
-    ]
-}
\ No newline at end of file
+    "exclude": ["node_modules", "**/tests/**", "**/fixtures/**", "**/*.test.ts", "**/e2e/**"]
```
Comment (Nodonisko) [inline packages/transport/tsconfig.lib.json:16]: why do we want exclude tests from type-check? I think we should check them too

File: packages/transport/src/lowlevel/sharedConnectionWorker.ts [hunk 5]
```diff
@@ -223,37 +162,92 @@ function handleAcquireIntent(
...
+    }
+    if (message.type === 'enumerate-done') {
+        handleReleaseDone(id); // port is the same as original
+    }
+}
+
+// TODO:
+// proper ts configuration for lib webworker
+// @ts-ignore
+if (typeof onconnect !== 'undefined') {
```
Comment (Nodonisko) [inline packages/transport/src/lowlevel/sharedConnectionWorker.ts:236]: not sure, but when you will have correct types for webworker in tsconfig, then you will probably still need to do `self.onconnect`

File: packages/transport/tsconfig.lib.json (hunk 4 above)
Comment (Nodonisko) [reply packages/transport/tsconfig.lib.json:16]: sorry I missed this is `.lib` so never mind

File: packages/transport/.eslintrc.js [hunk 6]
```diff
@@ -2,11 +2,10 @@ module.exports = {
     rules: {
         // todo: remove when refactoring transport layer.
```
Comment (Nodonisko) [inline packages/transport/.eslintrc.js:3]: maybe this could be removed when it is refactored now?

Comment (mroz22) [reply packages/transport/.eslintrc.js:3]: There are still 3 rules left 
'no-underscore-dangle': 'off',
'no-restricted-syntax': 'off',
'no-await-in-loop': 'off',

Removing these would mean touching runtime code to some greater extent and I didnt want to do that in this PR

File: packages/transport/src/fallback.ts (hunk 3 above)
Comment (mroz22) [reply packages/transport/src/fallback.ts:6]: this variable can't be assigned in constructor and is accessed multiple times in this file. If I wanted to define type here it would need to be `Transport | undefined` and in that case I would have to do some runtime check to every place where this is used or do something like `this.activeTransport!.call()...`
I dont know which one is better.

File: packages/transport/src/lowlevel/sharedConnectionWorker.ts (hunk 5 above)
Comment (mroz22) [reply packages/transport/src/lowlevel/sharedConnectionWorker.ts:236]: Here I decided to go with @ts-ignore for now as worker types were clashing with DOM types and I didnt find a quick way around it.

File: packages/transport/.eslintrc.js (hunk 2 above)
Comment (mroz22) [reply packages/transport/.eslintrc.js]: Ok, super, I just needed this for autofixing of backticks. Removing 29e9f6be8279856e6766466618a7e790fb802563

File: packages/transport/src/lowlevel/sharedConnectionWorker.ts (hunk 5 above)
Comment (Nodonisko) [reply packages/transport/src/lowlevel/sharedConnectionWorker.ts:236]: Ok, we can solve this later

**Review by Nodonisko — APPROVED** [review]

File: packages/integration-tests/projects/transport/common.setup.js (hunk 1 above)
Comment (mroz22) [reply packages/integration-tests/projects/transport/common.setup.js:22]: packages/integration-tests/projects/transport/tests/bridge.integration.js now runs against multiple bridge versions and it starts bridge in the test itself. See line 35. 
This setup is run in beforEach is it works like this:
- stop bridge
- do some setup
- start bridge version we want to test at the moment

Comment (Nodonisko) [reply packages/integration-tests/projects/transport/common.setup.js:22]: I see, thanks.
